### PR TITLE
Utility function to read and write directly to a bare repo

### PR DESCRIPTION
**Pull request by hoytak**
This PR adds simple utility functions to read and write files directly on a bare git repo.  

In this PR, these functions are not connected, but these are needed to enable some end-to-end tests that cover the different integration scenarios.

### CODE REVIEW

File: rust/gitxetcore/src/git_integration/git_wrap.rs [hunk 1]
```diff
@@ -353,3 +355,181 @@ pub fn perform_git_version_check() -> Result<()> {
...
+    // Create a tree from the blobs; do this in memory, then do it one by one.
+    let mut index = git2::Index::new()?;
+
+    for (file_name, file_oid, data_len) in file_oids {
+        let entry = git2::IndexEntry {
+            path: file_name.as_bytes().into(),
+            id: file_oid,
+            file_size: data_len as u32,
+            mode: 0o100644, // represents a blob (file)
+            dev: 0,
```
Comment (hoytak) [inline rust/gitxetcore/src/git_integration/git_wrap.rs:388]: From https://github.com/libgit2/libgit2/blob/45fd9ed7ae1a9b74b957ef4f337bc3c8b3df01b5/tests/libgit2/reset/hard.c#L105, these are fine to leave as zero.

File: rust/gitxetcore/src/git_integration/git_wrap.rs [hunk 2]
```diff
@@ -353,3 +355,181 @@ pub fn perform_git_version_check() -> Result<()> {
         }
     }
 }
+
+// Add files to a repo
```
Comment (seanses) [inline rust/gitxetcore/src/git_integration/git_wrap.rs:359]: Let's add a comment for the default behavior when branch_name is None.

File: rust/gitxetcore/src/git_integration/git_wrap.rs [hunk 3]
```diff
@@ -353,3 +355,181 @@ pub fn perform_git_version_check() -> Result<()> {
...
+        &signature,
+        &signature,
+        commit_message,
+        &tree,
+        &parents,
+    )?;
+
+    Ok(())
+}
+pub fn read_file_from_repo(
```
Comment (seanses) [inline rust/gitxetcore/src/git_integration/git_wrap.rs:439]: Also here a comment for branch is None.

**Review by seanses — APPROVED** [review]
LG!

**Pull request merged**